subworkflows/local/microfinder_map/main.nf

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env nextflow
+
+//
+// Adapted from MicroFinder.v0.2
+// by Tom Mathers
+//
+
+//
+// MODULE IMPORT BLOCK
+//
+
+include { MINIPROT_ALIGN        } from '../../../modules/nf-core/miniprot/align/main'
+include { MINIPROT_INDEX        } from '../../../modules/nf-core/miniprot/index/main'
+include { SORT_FASTA            } from '../../../modules/local/sort_fasta'
+include { MICROFINDER_FILTER    } from '../../../modules/local/microfinder_filter'
+include { RENAME_EMPTY_OUTPUT   } from '../../../modules/local/rename_empty_output/main'
+include { RENAME_SORTED_OUTPUT  } from '../../../modules/local/rename_sorted_output/main'
+
+// Function to check GFF content
+def checkGffContent(gff) {
+    def nonCommentLines = gff.readLines().findAll { !it.startsWith('#') }
+
+    if (nonCommentLines.isEmpty()) {
+        return [false, true]  // [hasContent, isEmpty]
+    }
+
+    def hasMrna = nonCommentLines.any { line ->
+        def fields = line.split('\t')
+        fields.size() >= 3 && fields[2] == "mRNA"
+    }
+
+    if (!hasMrna) {
+        return [false, true]  // If no mRNA entries, treat as empty
+    }
+
+    return [true, false]  // [hasContent, isEmpty]
+}
+
+workflow MICROFINDER_MAP {
+    take:
+    reference_tuple     // Channel: [ val(meta), path(file) ]
+    scaffold_length_cutoff // Channel: val(cutoff)
+    pep_files          // Channel: val(path or URL to protein file)
+    output_prefix      // Channel: val(prefix)
+
+    main:
+    ch_versions = Channel.empty()
+
+    //
+    // MODULE: CREATES INDEX OF REFERENCE FILE
+    //
+    MINIPROT_INDEX ( reference_tuple )
+    ch_versions = ch_versions.mix( MINIPROT_INDEX.out.versions )
+
+    //
+    // Create channel from pep_file and combine with index
+    //
+    def pep_ch = (pep_files instanceof groovyx.gpars.dataflow.DataflowReadChannel) ? pep_files : Channel.fromPath(pep_files)
+    pep_ch
+        .map { pf ->
+            def f = file(pf)
+            def meta = [ id: f.baseName, type: 'protein', org: 'reference' ]
+            tuple(meta, f)
+        }
```

Comment:
This would fit better in the initialisation sub-workflow

Reply:
also, initialisation doesn't accept mul-map, tried, it complained :(